maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/FdcContributionsService.java

```diff
@@ -141,27 +141,24 @@ private ContributionFilesEntity createFdcFile(CreateFdcFileRequest fdcFileReques
     @Transactional(rollbackFor =  MAATCourtDataException.class)
     @NotNull
     public Integer logFdcProcessed(LogFdcProcessedRequest request) {
-        FdcContributionsEntity fdcEntity = fdcContributionsRepository.findById(Integer.valueOf(request.getFdcId()))
+        FdcContributionsEntity fdcEntity = fdcContributionsRepository.findById(request.getFdcId())
                 .orElseThrow(() -> new RequestedObjectNotFoundException("fdc_contribution could not be found by id"));
         log.info("Contribution found: {}", fdcEntity.getId());
-        if (!debtCollectionService.updateContributionFileReceivedCount(fdcEntity.getContFileId())) {
-            throw new RequestedObjectNotFoundException("Contribution file could not be updated");
-        }
-        if(!StringUtils.isEmpty(request.getErrorText()) ){
+        if (!StringUtils.isEmpty(request.getErrorText())) {
             saveErrorMessage(request, fdcEntity);
+        } else if (!debtCollectionService.updateContributionFileReceivedCount(fdcEntity.getContFileId())) {
+            throw new NoSuchElementException("No associated contribution_file found for fdc_contribution");
```

Comment:
Likewise, include the Id in the exception message.

Reply:
done